src/ui/control/scroll_zoom_control.js

```diff
@@ -0,0 +1,322 @@
+        this._content.appendChild(htmlNode);
+        this._createCloseButton();
+        this._update();
+        return this;
+    }
+
+    /**
+     * Adds a CSS class to the scroll zoom control container element.
+     *
+     * @param {string} className Non-empty string with CSS class name to add to scroll zoom control container.
+     * @returns {ScrollZoomControl} Returns itself to allow for method chaining.
+     *
+     * @example
+     * const scrollZoomControl = new mapboxgl.ScrollZoomControl();
+     * scrollZoomControl.addClassName('some-class');
+     */
+    addClassName(className: string) {
+        this._classList.add(className);
+        if (this._container) {
+            this._updateClassList();
+        }
+        return this;
+    }
+
+    /**
+     * Removes a CSS class from the scroll zoom control container element.
+     *
+     * @param {string} className Non-empty string with CSS class name to remove from scroll zoom control container.
+     *
+     * @returns {ScrollZoomControl} Returns itself to allow for method chaining.
+     * @example
+     * const scrollZoomControl = new mapboxgl.ScrollZoomControl({className: 'some classes'});
+     * scrollZoomControl.removeClassName('some');
+     */
+    removeClassName(className: string) {
+        this._classList.delete(className);
+        if (this._container) {
+            this._updateClassList();
+        }
+        return this;
+    }
+
+    /**
+     * Add or remove the given CSS class on the scrollZoomControl container, depending on whether the container currently has that class.
+     *
+     * @param {string} className Non-empty string with CSS class name to add/remove.
+     *
+     * @returns {boolean} If the class was removed return `false`. If the class was added, then return `true`.
+     *
+     * @example
+     * const scrollZoomControl = new mapboxgl.ScrollZoomControl();
+     * scrollZoomControl.toggleClassName('highlighted');
+     */
+    toggleClassName(className: string) {
+        let finalState: boolean;
+        if (this._classList.delete(className)) {
+            finalState = false;
+        } else {
+            this._classList.add(className);
+            finalState = true;
+        }
+        if (this._container) {
+            this._updateClassList();
+        }
+        return finalState;
+    }
+
+    _createCloseButton() {
+        if (this.options.closeButton) {
+            this._closeButton = DOM.create('button', 'mapboxgl-scroll-zoom-control-close-button', this._content);
+            this._closeButton.type = 'button';
+            this._closeButton.setAttribute('aria-label', 'Close scroll zoom control alert');
+            this._closeButton.innerHTML = '&#215;';
+            this._closeButton.addEventListener('click', this._onClose);
+        }
+    }
+
+    _createFadeOut() {
+        const that = this;
+
+        if (this.options.closeFadeOut) {
+            setTimeout(() => {
```

Comment:
Can we use CSS transitions to avoid a timing loop, e.g.:

```css
...-container {
  opacity: 1;
  transition: opacity 300ms ease-in-out;
}
```

and then either set the opacity directly or toggle a class as needed? (I don't think we have to worry about `transitionend` events since nothing happens at the end of the transition.)